### ISSUE TITLE
Space causes one line to emit a warning and the other not

### ISSUE THREAD

**Issue by jackgerrits** [VowpalWabbit/vowpal_wabbit#3232]
#### Describe the bug
Only one of these lines produces a warning. The warning itself is kind of weird. What does it mean for an example to be input with no `|`. 

#### To Reproduce

data.txt:
```
a 
a
```
Note: line one has a trailing space, and line two does not.

Command line: `vw -d data.txt`

#### Expected behavior
Both lines produce the same warning irrespective of the trailing space.

#### Observed Behavior
Only one line produces a warning, the other is processed without warning.
```
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = data.txt
num sources = 1
Enabled reductions: gd, scorer-identity
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
[warning] warning: a is not a good float, replacing with 0
0.000000 0.000000            1            1.0   0.0000   0.0000        1
0.000000     n.a.            2            2.0  unknown   0.0000        1

finished run
number of examples = 2
weighted example sum = 2.000000
weighted label sum = 0.000000
average loss = 0.000000
total feature number = 2
```

#### Environment
What version of VW did you use? c02ef867914967dc4dc4096e33aeb619b63e8f75
What OS or language did you use? Ubuntu, command line

**Comment by jackgerrits** [comment]
I think this is expected, although seems like a bad user experience.

If there is no `|` it is assumed that the input line is the label (bit that comes before the `|`). The tag is the last token if there is no whitespace between it and the `|`. Therefore, for the line with a space at the end the `a` is parsed as a simple label which clearly fails to be parsed as a float.

For the second one with no whitespace at the end it is interpreted as a tag. It is saved into the example and then removed from the line. The result is no simple label to be parsed.
